lilypad/_opentelemetry/_opentelemetry_vertex/utils.py

```diff
@@ -0,0 +1,194 @@
+from typing import Any
+
+from opentelemetry.semconv._incubating.attributes import gen_ai_attributes
+from opentelemetry.trace import Span
+
+
+def get_vertex_model_name(instance: Any) -> str:
+    # Attempt to extract model name from instance of GenerativeModel.
+    llm_model = "unknown"
+    if hasattr(instance, "_model_name") and instance._model_name:
+        llm_model = instance._model_name
+    return llm_model
+
+
+def get_vertex_llm_request_attributes(
+    kwargs: dict[str, Any], instance: Any
+) -> dict[str, Any]:
+    attributes: dict[str, Any] = {
+        gen_ai_attributes.GEN_AI_OPERATION_NAME: "chat",
+        gen_ai_attributes.GEN_AI_SYSTEM: "vertex",
```

Comment:
https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-metrics/

this page gets updated pretty frequently (they keep adding new providers).

update the opentelemetry package as necessary and use the providers from opentelemetry package.